openquake/sep/liquefaction/liquefaction.py

```diff
@@ -113,18 +153,488 @@ def zhu_liquefaction_probability_general(
+    :param cti:
+        Compound Topographic Index, a proxy for soil wetness.
+    :param vs30:
+        Shear-wave velocity averaged over the upper 30 m of the earth at the
+        site.
+
+    :returns:
+        prob_liq: Probability of liquefaction at the site.
+        out_class: Binary output 0 or 1, i.e., liquefaction nonoccurrence
+                   or liquefaction occurrence occurrence.
+    """
+    pga_scale = pga * _idriss_magnitude_weighting_factor(mag)
+    Xg = (
+        pgam_coeff * np.log(pga_scale)
+        + cti_coeff * cti
+        + vs30_coeff * np.log(vs30)
+        + intercept
+    )
+    prob_liq = sigmoid(Xg)
+    out_class = np.where(prob_liq > 0.57, 1, 0)
+    return prob_liq, out_class
+
+
+def todorovic_silva_2022_nonparametric_general(
+    pgv: Union[float, np.ndarray],
+    vs30: Union[float, np.ndarray],
+    dw: Union[float, np.ndarray],
+    wtd: Union[float, np.ndarray],
+    precip: Union[float, np.ndarray],
+    ) -> Union[float, np.ndarray]:
+    """
+    Returns the binary class output (i.e, 0 or 1) which indicates liquefaction
+    nonoccurrence or liquefaction occurrence as per Todorovic and Silva (2022). 
+    In addition, it returns the probability of belonging to a class 1, or liq-
+    uefaction occurrence. The implemented model includes modifications from the 
+    version published in Todorovic and Silva (2022).
+
+    Reference: Todorovic, L., Silva, V. (2022). 
+    A liquefaction occurrence model for regional analysis. 
+    Soil Dynamics and Earthquake Engineering, 161, 1–12. 
+    https://doi.org/10.1016/j.soildyn.2022.107430
+
+    :param pgv:
+        Peak Ground Velocity, measured in cm/s
+    :param vs30:
+        Shear-wave velocity averaged over the upper 30 m of the earth at the
+        site, measured in m/s
+    :param dw:
+        Distance to the nearest water body, measured in km
+    :param wtd:
+        Global water table depth, measured in m
+    :param precip:
+        Mean annual precipitation, measured in mm
+
+    :returns:
+        out_class: ndarray of len(sitemodel)  
+            Returns binary output 0 or 1, i.e., liquefaction nonoccurrence or
+            liquefaction occurrence occurrence.
+        out_prob: ndarray of len(sitemodel)
+            Returns probability of belonging to class 1.
+        d32b421
+
+    """
+    strain_proxy = pgv / (CM_PER_M * vs30)
+    matrix = np.array([strain_proxy, dw, wtd, precip]).T
+    
+    model_file = 'data/todorovic_silva_2022/todorovic_silva_2022.onnx.gz'
+    model_path = os.path.join(os.path.dirname(__file__), model_file)
+    with gzip.open(model_path, 'rb') as gzipped_file:
+        file = gzipped_file.read()
+        providers = ['CPUExecutionProvider', 'AzureExecutionProvider']
+        session = onnxruntime.InferenceSession(file, providers=providers)
+        out_class = session.run(None, {"X": matrix})[0]
+        out_prob = [_[1] for _ in session.run(None, {"X":matrix})[1]]
```

Comment:
Can avoid computing the predictions from the model twice by capturing the results returned on the first call: 
```
res = session.run(None, {"X": matrix})
out_class = res[0]
out_prob = [p[1] for p in res[1]]
```

Reply:
Fixed.